### PR TITLE
When Bridge Aggregation Is Disabled, Prevent Borealis Memory Leak

### PR DESCRIPTION
**Pull request by matt001k**
When transmit aggregations are disabled on the bridge, memory is leaked into the spectral entropy list whenever a Borealis levels message is received.
The aggregation method in the BorealisSensor class will never get called and the entropy list elements will never be freed.

## What changed?
Only allow levels SPL data to be added to spectral entropy list when aggregations are enabled
Always clear spectral entropy list whenever an aggregation report is created

## How does it make Bristlemouth better?
When bridge report aggregation parameter is disabled, prevent memory from leaking through the spectral entropy list
Always clears the spectral entropy list whenever an aggregation report is created preventing memory from never being freed when aggregation reporting is enabled

## Where should reviewers focus?
Any other edge cases that could be leaking memory here?
From my initial testing everything looks good, but would like more eyes on this file.

## Checklist

- [ ] Add or update unit tests for changed code
- [ ] Ensure all submodules up to date. If this PR relies on changes in submodules, merge those PRs first, then point this PR at/after the merge commit
- [ ] Ensure code is formatted correctly with clang-format. If there are large formatting changes, they should happen in a separate whitespace-only commit on this PR after all approvals.

### CODE REVIEW

File: src/apps/bridge/sensor_drivers/borealisSensor.cpp [hunk 1]
```diff
@@ -229,6 +227,8 @@ void BorealisSensor::aggregate(void) {
             exceedsConfiguredThresholds(report.spl, report.max_iqr, report.entropy);
       }
     }
+    // Clear the spectral entropy list after entropy has been calculated
+    clear_spectral_entropy_list();
```
Comment (matt001k) [inline src/apps/bridge/sensor_drivers/borealisSensor.cpp:231]: If line 212 is false because we cannot allocate memory, then the spectral entropy list is never freed.
This is a very plausible thing to occur, if the `report_interval` on Borealis is set higher than the `sampleDurationMs` of the bridge, then the `bm_malloc` will fail due to an attempt to allocate 0 bytes (evaluate the function definition of `pvPortMalloc` and see that if 0 bytes are attempted to be allocated, it will return NULL).
It is best to free the memory here!

File: src/apps/bridge/sensor_drivers/borealisSensor.cpp [hunk 2]
```diff
@@ -476,6 +479,7 @@ void BorealisSensor::borealisSubCallback(uint64_t node_id, const char *topic,
             // to be less than half of the bridge power controller's sampleDurationMs, we cannot
             // send multiple of level stat messages yet and need to free the unused string
             bm_free(borealis->tracking_data.stats.levels);
+            borealis->tracking_data.stats.levels = NULL;
```
Comment (matt001k) [inline src/apps/bridge/sensor_drivers/borealisSensor.cpp:482]: NULL this to prevent a potential double free if the `report_interval` ever gets changed.

File: src/lib/dsp/spectral_entropy.c [hunk 3]
```diff
@@ -90,11 +90,9 @@ void clear_spectral_entropy_list(void) {
   }
 }
 
-float calc_min_spectral_entropy_and_clear_list(const unsigned int num_bands,
-                                               float const *const means) {
+float calc_min_spectral_entropy(const unsigned int num_bands, float const *const means) {
   min_entropy = 1.0f;
   SpectralEntropyArgs arg = {.num_bands = num_bands, .means = means};
   ll_traverse(&spectra, compute_entropies_and_find_minimum, &arg);
-  clear_spectral_entropy_list();
```
Comment (matt001k) [inline src/lib/dsp/spectral_entropy.c:98]: This is now cleared in the data aggregation method.

**Review by towynlin — APPROVED** [review]
Great work!

**Review by victorsowa12 — APPROVED** [review]
